src/fireedge/src/modules/utils/parser/vmTemplateFilter.js

```diff
@@ -0,0 +1,810 @@
+        filterSingleSection(formData, correctionMap, section, newExtra, key)
+
+        // Special cases for Context.USER_INPUTS
+        if (
+          section === 'Context' &&
+          key === 'USER_INPUTS' &&
+          correctionMap.extra.Context.USER_INPUTS
+        ) {
+          // Keep CONTEXT.INPUTS_ORDER because it's not a form by himself, depends on CONTEXT.USER_INPUTS
+          newExtra.INPUTS_ORDER = formData.extra.INPUTS_ORDER
+        }
+      }
+    })
+  }
+}
+
+/**
+ * Filter one section that is not Storage, Network or Schedule Actions.
+ *
+ * @param {object} formData - The data from the form (the one that the user could be modify)
+ * @param {object} itemsModifications - Map with the fields that will be touched and modified by the user
+ * @param {string} section - Section of the form
+ * @param {object} newExtra - The extra section of the form.
+ * @param {string} key - Section on the formData
+ */
+const filterSingleSection = (
+  formData,
+  itemsModifications,
+  section,
+  newExtra,
+  key
+) => {
+  // Check if attribute has changes on the correction map
+  if (
+    (key in formData.extra || key in itemsModifications.extra[section]) &&
+    itemsModifications.extra[section]?.[key]
+  ) {
+    // Get value and copy the section
+    const value = formData.extra[key]
+    const newOtherSection = { ...newExtra[key] }
+
+    // Arrays and single values replace the whole value
+    if (
+      Array.isArray(value) ||
+      key === 'USER_INPUTS' ||
+      !(typeof value === 'object')
+    ) {
+      newExtra[key] = value
+    } else {
+      // Objects iterate over each key to check if the key was changed by the user
+      Object.entries(itemsModifications.extra[section]?.[key] || {}).forEach(
+        ([childrenKey, correction]) => {
+          // If the correction is boolean, means that the user changed this value that is a simple value
+          // If the correction is an object with an attribute delete means that is a hidden field that was deleted because the user change the value of its parent
+          // If the correction is an object without an attribute delete means that is a field with an object instead a simple value and the user changed its value
+          if (
+            correction &&
+            typeof correction === 'object' &&
+            correction.__delete__
+          ) {
+            delete newOtherSection[childrenKey]
+          } else if (
+            correction &&
+            typeof correction === 'boolean' &&
+            childrenKey in formData.extra[key] &&
+            (!_.isEmpty(formData.extra[key][childrenKey]) ||
+              formData.extra[key][childrenKey] !== null)
+          ) {
+            newOtherSection[childrenKey] = formData.extra[key][childrenKey]
+          } else if (
+            correction &&
+            typeof correction === 'boolean' &&
+            !(childrenKey in formData.extra[key])
+          ) {
+            delete newOtherSection[childrenKey]
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We detected that you are using `delete` to remove items from an array. If we use `delete`, it will delete the object property, but will not reindex the array or update its length. It leaves `undefined` holes in the array. Therefore, we recommend you to use `splice` to remove items from an array.

[Learn more](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete)